File: Estudos solo/Python/exercicios atividade 7 em python/at7ex1.py

def somaDigitos(x):
    if(x < 10):
        soma = x
        return soma
    else:
        soma = x%10 + somaDigitos((x-x%10)/10)
        return soma

numero = int(input("Numero: "))

print(somaDigitos(numero))

File: ATP 2/Codigos/C e C++/AULA 8/at7_ex1.cpp
#include <stdio.h>
#include <stdlib.h>
#include <ctype.h>

int soma = 0;

int soma_digitos (int x)
{
    if(x < 10)
    {
        soma += x;
        return soma;
    }
    soma += x%10;
    return(soma_digitos(x/10));
}

int main()
{
    printf("\n%d", soma_digitos(123));

    getchar();
    return 0;
}
